src/Algebra/Polynomial/Morphism/Morphism.agda

```diff
@@ -0,0 +1,510 @@
+    eval-p (a ∷ p)
+      ≈⟨ B.+-congʳ (S.IsRelHomomorphism.cong
+         (RingHomomorphism.isRelHomomorphism f) (head (≈0⇒IsZero p≈q))) ⟩
+    ⟦ A0# ⟧ +B β *B eval-p p
+      ≈⟨ B.+-congˡ (B.*-congˡ (eval-p-cong {p = p} {q = []}
+         (IsZero⇒≈0 (tail (≈0⇒IsZero p≈q))))) ⟩
+    ⟦ A0# ⟧ +B β *B eval-p []
+      ≈⟨ B.+-congʳ (MonoidHomomorphism.ε-homo
+         (RingHomomorphism.+-monoidHomomorphism f)) ⟩
+    0# +B β *B eval-p []
+      ≈⟨ B.+-congˡ (B.zeroʳ β) ⟩
+    0# +B 0#
+      ≈⟨ B.+-identityʳ 0# ⟩
+    eval-p []
+      ∎
+  eval-p-cong {p = a ∷ p} {b ∷ q} (cons a≈b p≈q)
+    = B.+-cong (S.IsRelHomomorphism.cong (RingHomomorphism.isRelHomomorphism f) a≈b)
+      (B.*-congˡ (eval-p-cong p≈q))
+
+
+  eval-cong : P ≈ Q → eval P ≈B eval Q
+  eval-cong {P = (m , p)} {Q = (n , q)} P≈Q = begin
+    eval (m , p)
+      ≈⟨ eval-equiv p ⟨
+    eval-p p
+      ≈⟨ eval-p-cong P≈Q ⟩
+    eval-p q
+      ≈⟨ eval-equiv q ⟩
+    eval (n , q)
+      ∎
+
+
+
+  eval-p-+-homomorphic : (p : Poly m) → (q : Poly n) → eval-p (p +P q) ≈B eval-p p +B eval-p q
+  eval-p-+-homomorphic [] q = begin
+    eval-p ([] +P q)
+      ≈⟨ eval-p-cong (+-P-identityˡ {m = 0} q) ⟩
+    eval-p q
+      ≈⟨ B.+-identityˡ (eval-p q) ⟨
+    eval-p [] +B eval-p q
+      ∎
+  eval-p-+-homomorphic (a ∷ p) [] = B.sym (B.+-identityʳ (eval-p (a ∷ p)))
+  eval-p-+-homomorphic (a ∷ p) (b ∷ q) = begin
+    eval-p ((a ∷ p) +P (b ∷ q))
+      ≈⟨ B.+-congˡ (B.*-congˡ (eval-p-+-homomorphic p q)) ⟩
+    ⟦ a +A b ⟧ +B (β *B (eval-p p +B eval-p q))
+      ≈⟨ B.+-congˡ (proj₁ distrib β (eval-p p) (eval-p q)) ⟩
+    ⟦ a +A b ⟧ +B (β *B eval-p p +B β *B eval-p q)
+      ≈⟨ B.+-congʳ (IsMagmaHomomorphism.homo (RingHomomorphism.+-isMagmaHomomorphism f) a b) ⟩
+    ⟦ a ⟧ +B ⟦ b ⟧ +B (β *B eval-p p +B β *B eval-p q) 
+      ≈⟨ B.+-assoc (⟦ a ⟧ +B ⟦ b ⟧) (β *B eval-p p) (β *B eval-p q) ⟨
+    ⟦ a ⟧ +B ⟦ b ⟧ +B β *B eval-p p +B β *B eval-p q
+      ≈⟨ B.+-congʳ (B.+-assoc ⟦ a ⟧ ⟦ b ⟧ (β *B eval-p p)) ⟩
+    ⟦ a ⟧ +B (⟦ b ⟧ +B β *B eval-p p) +B β *B eval-p q
+      ≈⟨ B.+-assoc  (⟦ a ⟧) (⟦ b ⟧ +B β *B eval-p p) (β *B eval-p q) ⟩
+    ⟦ a ⟧ +B ( ⟦ b ⟧ +B β *B eval-p p +B (β *B eval-p q))
+      ≈⟨ B.+-congˡ (B.+-congʳ ((B.+-comm (β *B eval-p p) ⟦ b ⟧))) ⟨
+    ⟦ a ⟧ +B ((β *B eval-p p) +B ⟦ b ⟧ +B (β *B eval-p q))
+      ≈⟨ B.+-congˡ (B.+-assoc (β *B eval-p p) ⟦ b ⟧ (β *B eval-p q)) ⟩
+    ⟦ a ⟧ +B (β *B eval-p p +B eval-p (b ∷ q))
+      ≈⟨ B.+-assoc ⟦ a ⟧ (β *B eval-p p) (eval-p (b ∷ q)) ⟨ 
+    ⟦ a ⟧ +B β *B eval-p p +B eval-p (b ∷ q)
+      ∎
+
+
+
+  eval-+-homomorphic : (P : Polynomial) → (Q : Polynomial) → eval (P + Q) ≈B eval P +B eval Q
+  eval-+-homomorphic (m , p) (n , q) = begin
+    eval ((m , p) + (n , q))
+      ≈⟨ eval-equiv (p +P q) ⟨
+    eval-p (p +P q)
+      ≈⟨ eval-p-+-homomorphic p q ⟩
+    eval-p p +B eval-p q
+      ≈⟨ B.+-cong (eval-equiv p) (eval-equiv q) ⟩
+    eval (m , p) +B eval (n , q)
+      ∎
+
+  eval-p-1#-homomorphic : eval-p (A1# ∷ []) ≈B 1#
```

Comment:
Inlining the definition of a constant polynomial makes this definition harder to read. Plus this is true for any member of the ground ring.